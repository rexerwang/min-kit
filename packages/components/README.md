# @min-kit/components

[![](https://img.shields.io/codecov/c/github/rexerwang/min-kit?flag=components&style=for-the-badge)](https://codecov.io/gh/rexerwang/min-kit/flags)
[![](https://img.shields.io/npm/types/%40min-kit/components?style=for-the-badge)](https://github.com/rexerwang/min-kit/tree/main/packages/components)
[![](https://img.shields.io/npm/v/%40min-kit/components?style=for-the-badge)](https://npm.im/@min-kit/components)
[![](https://img.shields.io/badge/React-Tarojs-007ACC?style=for-the-badge&logo=react&logoColor=61DAFB&labelColor=20232A)](https://github.com/NervJS/taro)

React components for miniapp

## Usage

```tsx
import { MinIcon, MinImage, MinNavigationBar, Modal, type ReactProps } from '@min-kit/components'
import { go } from '@min-kit/extends'
import { Button, View } from '@tarojs/components'

// pages/index/index.tsx
export default function Index(props: ReactProps) {
  const openModal = () => {
    Modal.confirm({
      title: 'Confirm',
      subtitle: 'the subtitle\nsupport multiple lines.',
      okText: 'ok',
      cancelText: 'cancel',
    })
  }

  return (
    <View className='relative box-border'>
      <MinNavigationBar
        title={
          <View className='flex-center-y'>
            <MinIcon name='back' size={20} onClick={() => go.back()} />
            <View className='ml-3 text-sm'>@min-kit/components</View>
          </View>
        }></MinNavigationBar>

      <View className='flex flex-col items-center'>
        <MinImage
          className='w-[750px] h-[750px]'
          src='https://source.unsplash.com/random/400x400?nature'
          preview={[
            'https://source.unsplash.com/random/400x400?nature',
            'https://source.unsplash.com/random/400x400?people',
            'https://source.unsplash.com/random/400x400?foods',
          ]}
        />
        <Button className='mt-4' type='primary' size='mini' plain onClick={openModal}>
          open modal
        </Button>
      </View>
    </View>
  )
}

definePageConfig({
  navigationStyle: 'custom',
})
```

### Import styles

```js
// Import overall styles
import '@min-kit/components/dist/styles/index.css'

// Import individual component styles
import '@min-kit/components/dist/styles/min-debugger.css'
import '@min-kit/components/dist/styles/min-icon.css'
import '@min-kit/components/dist/styles/min-image.css'
import '@min-kit/components/dist/styles/min-navigation-bar.css'
import '@min-kit/components/dist/styles/modal.css'

// Import SCSS
import '@min-kit/components/styles/index.scss'
import '@min-kit/components/styles/min-debugger.scss'
import '@min-kit/components/styles/min-icon.scss'
import '@min-kit/components/styles/min-image.scss'
import '@min-kit/components/styles/min-navigation-bar.scss'
import '@min-kit/components/styles/modal.scss'
```

More usage examples ðŸ‘‰ https://github.com/rexerwang/min-kit/tree/main/packages/example
