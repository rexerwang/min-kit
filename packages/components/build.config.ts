import { exec } from 'node:child_process'
import { resolve } from 'node:path'
import { promisify } from 'node:util'

import { defineBuildConfig } from 'unbuild'

export default defineBuildConfig({
  clean: !process.argv.includes('--no-clean'),
  entries: [
    'src/index',
    {
      builder: 'mkdist',
      input: 'styles',
      outDir: 'dist/styles',
    },
  ],
  declaration: true,
  rollup: {
    emitCJS: true,
    esbuild: {
      jsx: 'automatic',
    },
  },
  hooks: {
    async 'mkdist:done'() {
      const input = resolve(__dirname, 'dist/styles')
      await promisify(exec)(`pnpm postcss ${input} -d ${input}`)
    },
  },
})
