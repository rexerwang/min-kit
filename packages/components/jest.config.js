module.exports = require('@min-kit/jest/config')({
  collectCoverageFrom: ['!src/debugger/{hooks,service,store}/**'],
})
