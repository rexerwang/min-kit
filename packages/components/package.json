{
  "name": "@min-kit/components",
  "version": "0.5.1",
  "description": "React components for miniapp",
  "author": "rexerwang",
  "license": "MIT",
  "repository": "rexerwang/min-kit",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.mjs",
      "require": "./dist/index.cjs"
    },
    "./dist/styles/*.css": {
      "import": "./dist/styles/*.css",
      "require": "./dist/styles/*.css"
    }
  },
  "main": "./dist/index.cjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "files": [
    "dist",
    "styles"
  ],
  "keywords": [
    "min-kit",
    "components",
    "react",
    "taro",
    "miniapp",
    "miniprogram"
  ],
  "scripts": {
    "build": "unbuild",
    "test": "jest"
  },
  "dependencies": {
    "@min-kit/extends": "workspace:*",
    "@min-kit/hooks": "workspace:*",
    "@min-kit/shared": "workspace:*",
    "@min-kit/store": "workspace:*",
    "clsx": "^2.1.0"
  },
  "peerDependencies": {
    "@tarojs/components": ">=3.6",
    "@tarojs/plugin-platform-alipay": ">=3.6",
    "@tarojs/plugin-platform-h5": ">=3.6",
    "@tarojs/plugin-platform-jd": ">=3.6",
    "@tarojs/plugin-platform-qq": ">=3.6",
    "@tarojs/plugin-platform-swan": ">=3.6",
    "@tarojs/plugin-platform-tt": ">=3.6",
    "@tarojs/plugin-platform-weapp": ">=3.6",
    "@tarojs/react": ">=3.6",
    "@tarojs/taro": ">=3.6",
    "react": ">=17",
    "react-dom": ">=17"
  },
  "devDependencies": {
    "@min-kit/helper": "workspace:*",
    "@min-kit/jest": "workspace:^"
  }
}
