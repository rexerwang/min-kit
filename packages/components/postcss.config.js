module.exports = {
  plugins: {
    tailwindcss: {},
    'postcss-rem-to-responsive-pixel': {
      rootValue: 32,
      propList: ['*'],
      transformUnit: 'rpx',
    },
  },
}
