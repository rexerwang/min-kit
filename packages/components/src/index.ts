export * from './debug'
export * from './gesture'
export * from './icon'
export * from './img'
export * from './modal'
export * from './modal/drawer'
export * from './navigation-bar'
export * from './portal'
export * from './share'
export * from './text'
export type { ReactProps } from './types'
