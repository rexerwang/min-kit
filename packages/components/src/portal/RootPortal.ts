import { RootPortal, View } from '@tarojs/components'
import { canIUse } from '@tarojs/taro'

// TODO: fix RootPortal not defined in some platforms (tt, ...)
export default canIUse('root-portal') ? RootPortal : View
