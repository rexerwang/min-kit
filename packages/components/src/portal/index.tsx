import { document, getRootElement, logger } from '@min-kit/extends'
import { RootPortal, View } from '@tarojs/components'
import { render, unmountComponentAtNode } from '@tarojs/react'
import { canIUse } from '@tarojs/taro'

const Portal = canIUse('root-portal') ? RootPortal : View

let n = 0

export type UnmountPortal = {
  (): boolean
  id: string
}

/**
 * mount Portal on current page instance
 * @throws if not found page instance
 */
export function mountPortal<T extends object>(Component: React.FC<T>, props: T = {} as T, displayName?: string) {
  const meta = {
    id: '',
    name: displayName ?? Component.displayName ?? Component.name,
    props,
  }

  const root = getRootElement()

  if (!root) {
    const error = new Error('Not found page instance')
    logger.error('#mountPortal', 'Mount', error, meta)
    throw error
  }

  const el = document.createElement('block')
  const id = 'min-portal-' + ++n

  meta.id = id

  render(
    <Portal id={id} className='min-portal'>
      <Component {...props} />
    </Portal>,
    el,
  )

  root.appendChild(el)
  logger.debug('#mountPortal', 'Mount', meta)

  const unmount: UnmountPortal = () => {
    try {
      unmountComponentAtNode(el)
      root.removeChild(el)
      logger.debug('#mountPortal', 'Unmount', meta)
      return true
    } catch (error) {
      logger.error('#mountPortal', 'Unmount', error, meta)
      return false
    }
  }
  unmount.id = id
  return unmount
}
