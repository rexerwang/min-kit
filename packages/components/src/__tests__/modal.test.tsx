import { act, fireEvent, render, screen, spyOnConsole } from '@min-kit/jest'
import { View } from '@tarojs/components'
import { createPortal } from 'react-dom'

import * as portals from '../portal/index'

import type { IModal } from '../modal/types'

const ModalStub = (props: IModal.Props) => (
  <View data-testid='test'>
    <View>content</View>
    <View onClick={props.onOk}>ok</View>
    <View onClick={props.onCancel}>cancel</View>
  </View>
)

describe('modal', () => {
  spyOnConsole() // disable console

  const root = document.createElement('page')

  let spy: jest.SpyInstance
  beforeEach(() => {
    spy = jest.spyOn(portals, 'mountPortal').mockImplementation((Component, props): any => {
      const el = document.createElement('root-portal')
      el.classList.add('portal')
      const { unmount } = render(createPortal(<Component {...props} />, el))
      root.appendChild(el)
      document.body.appendChild(root)

      return () => {
        try {
          unmount()
          root.removeChild(el)
          document.body.removeChild(root)
          return true
        } catch {
          return false
        }
      }
    })
  })

  describe('Modal.with', () => {
    it('should render modal toMatchSnapshot', async () => {
      const { Modal } = await import('../modal/index')
      const modal = Modal.with('ModalStub', ModalStub)()
      expect(root).toMatchSnapshot()
      modal.unmount()
    })

    it('should close modal with `ok = true` when ok button clicked', async () => {
      const { Modal } = await import('../modal/index')
      const modal = Modal.with('ModalStub', ModalStub)()

      act(() => {
        fireEvent.click(screen.getByText(/ok/))
      })

      await expect(modal).resolves.toEqual(expect.objectContaining({ ok: true }))
    })

    it('should close modal with `ok = false` when cancel button clicked', async () => {
      const { Modal } = await import('../modal/index')
      const modal = Modal.with('ModalStub', ModalStub)()

      act(() => {
        fireEvent.click(screen.getByText(/cancel/))
      })

      await expect(modal).resolves.toEqual(expect.objectContaining({ ok: false }))
    })

    it('should close modal with `ok = false` when backdrop clicked', async () => {
      const { Modal } = await import('../modal/index')
      const modal = Modal.with('ModalStub', ModalStub)({ backdropCloseable: true })

      act(() => {
        fireEvent.click(root.querySelector('.min-drawer')!)
      })

      await expect(modal).resolves.toEqual({ ok: false, detail: 'backdrop' })
    })

    it('should close modal with `ok = false` when unmount', async () => {
      const { Modal } = await import('../modal/index')
      const modal = Modal.with('ModalStub', ModalStub)({ backdropCloseable: true })

      act(() => {
        modal.unmount()
      })

      await expect(modal).resolves.toEqual({ ok: false, detail: 'unmount' })
    })

    it('should return false when unmount duplicated', async () => {
      const { Modal } = await import('../modal/index')
      const modal = Modal.with('ModalStub', ModalStub)({ backdropCloseable: true })

      act(() => {
        modal.unmount()
      })

      expect(modal.unmount()).toBeFalsy()
    })

    it('should thrown when mountPortal with exception', async () => {
      spy.mockReset().mockImplementation(() => {
        throw new Error('')
      })

      const { Modal } = await import('../modal/index')
      await expect(Modal.with('ModalStub', ModalStub)()).rejects.toThrow()
    })

    it('should render modal with top position', async () => {
      const { Modal } = await import('../modal/index')
      const modal = Modal.with(
        'ModalStub',
        ModalStub,
      )({
        position: 'top',
      })

      expect(root.querySelector('.top')).toBeInTheDocument()

      modal.unmount()
    })

    it('should render modal with bottom position', async () => {
      const { Modal } = await import('../modal/index')
      const modal = Modal.with(
        'ModalStub',
        ModalStub,
      )({
        position: 'bottom',
      })

      expect(root.querySelector('.bottom')).toBeInTheDocument()

      modal.unmount()
    })
  })

  describe('Modal.confirm render toMatchSnapshot', () => {
    it('should render Modal.confirm toMatchSnapshot', async () => {
      const { Modal } = await import('../modal/index')

      const confirm = Modal.confirm({
        icon: 'icon',
        title: 'Modal.confirm',
        subtitle: 'confirm subtitle',
        okText: 'ok',
        footer: 'confirm footer',
      })

      expect(root).toMatchSnapshot()
      confirm.unmount()
    })

    it('should render Modal.confirm with cancel button', async () => {
      const { Modal } = await import('../modal/index')

      const confirm = Modal.confirm({
        icon: 'icon',
        title: 'Modal.confirm',
        subtitle: 'confirm subtitle',
        okText: 'ok',
        cancelText: 'cancel',
        footer: 'confirm footer',
      })

      expect(root).toMatchSnapshot()
      confirm.unmount()
    })

    it('should render Modal.confirm with buttons inline layout', async () => {
      const { Modal } = await import('../modal/index')

      const confirm = Modal.confirm({
        icon: 'icon',
        title: 'Modal.confirm',
        subtitle: 'confirm subtitle',
        okText: 'ok',
        cancelText: 'cancel',
        buttonInline: true,
        footer: 'confirm footer',
      })

      expect(root).toMatchSnapshot()
      confirm.unmount()
    })

    it('should render Modal.confirm with custom icon & no-divider', async () => {
      const { Modal } = await import('../modal/index')

      const confirm = Modal.confirm({
        icon: <View className='custom-icon'>icon</View>,
        title: 'Modal.confirm',
        okText: 'ok',
        divider: false,
      })

      expect(root).toMatchSnapshot()
      confirm.unmount()
    })
  })

  describe('Modal.confirm with events', () => {
    it('should close confirm with `ok = true` when ok button clicked', async () => {
      const { Modal } = await import('../modal/index')

      const confirm = Modal.confirm({
        title: 'Modal.confirm',
        okText: 'ok',
      })

      act(() => {
        fireEvent.click(screen.getByText(/ok/))
      })

      await expect(confirm).resolves.toEqual({ ok: true, detail: undefined })
    })

    it('should close confirm with `ok = true` when ok button clicked with inline layout', async () => {
      const { Modal } = await import('../modal/index')

      const confirm = Modal.confirm({
        title: 'Modal.confirm',
        okText: 'ok',
        cancelText: 'cancel',
        buttonInline: true,
      })

      act(() => {
        fireEvent.click(screen.getByText(/ok/))
      })

      await expect(confirm).resolves.toEqual({ ok: true, detail: undefined })
    })

    it('should close confirm with `ok = false` when cancel button clicked', async () => {
      const { Modal } = await import('../modal/index')

      const confirm = Modal.confirm({
        title: 'Modal.confirm',
        okText: 'ok',
        cancelText: 'cancel',
      })

      act(() => {
        fireEvent.click(screen.getByText(/cancel/))
      })

      await expect(confirm).resolves.toEqual({ ok: false, detail: 'cancel' })
    })

    it('should close confirm with `ok = false` when ok button clicked with inline layout', async () => {
      const { Modal } = await import('../modal/index')

      const confirm = Modal.confirm({
        title: 'Modal.confirm',
        okText: 'ok',
        cancelText: 'cancel',
        buttonInline: true,
      })

      act(() => {
        fireEvent.click(screen.getByText(/cancel/))
      })

      await expect(confirm).resolves.toEqual({ ok: false, detail: 'cancel' })
    })

    it('should close confirm with `ok = false` when close icon clicked', async () => {
      const { Modal } = await import('../modal/index')

      const confirm = Modal.confirm({
        title: 'Modal.confirm',
        okText: 'ok',
        cancelText: 'cancel',
        closeable: true,
      })

      act(() => {
        fireEvent.click(root.querySelector('.close > *')!)
      })

      await expect(confirm).resolves.toEqual({ ok: false, detail: 'close' })
    })
  })
})
