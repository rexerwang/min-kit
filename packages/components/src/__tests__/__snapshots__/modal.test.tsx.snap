// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal Modal.confirm render toMatchSnapshot should render Modal.confirm toMatchSnapshot 1`] = `
<page>
  <root-portal
    class="portal"
  >
    <taro-view-core
      class="min-drawer center"
      style="z-index: 1008; --duration: 300ms; --offsetX: 0px; --offsetY: -96px;"
    >
      <!---->
      <taro-view-core
        class="min-drawer-body"
      >
        <!---->
        <taro-view-core
          class="min-confirm-modal"
        >
          <!---->
          <taro-image-core
            class="icon"
            style="width: 60px; height: 60px;"
          />
          <taro-view-core
            class="title"
          >
            <!---->
            <taro-text-core
              class=""
              decode="true"
              space="nbsp"
            >
              <!---->
              Modal.confirm
            </taro-text-core>
          </taro-view-core>
          <taro-view-core
            class="subtitle"
          >
            <!---->
            <taro-text-core
              class=""
              decode="true"
              space="nbsp"
            >
              <!---->
              confirm subtitle
            </taro-text-core>
          </taro-view-core>
          <taro-view-core
            class="divider"
          >
            <!---->
          </taro-view-core>
          <taro-view-core
            class="buttons"
          >
            <!---->
            <taro-button-core
              class="button ok"
            >
              <!---->
              ok
            </taro-button-core>
          </taro-view-core>
          <taro-view-core
            class="footer"
          >
            <!---->
            <taro-text-core
              class=""
              decode="true"
              space="nbsp"
            >
              <!---->
              confirm footer
            </taro-text-core>
          </taro-view-core>
          <taro-view-core
            class="close"
          >
            <!---->
            <taro-image-core
              class=""
              style="width: 24px; height: 24px;"
            />
          </taro-view-core>
        </taro-view-core>
      </taro-view-core>
    </taro-view-core>
  </root-portal>
</page>
`;

exports[`modal Modal.confirm render toMatchSnapshot should render Modal.confirm with buttons inline layout 1`] = `
<page>
  <root-portal
    class="portal"
  >
    <taro-view-core
      class="min-drawer center"
      style="z-index: 1010; --duration: 300ms; --offsetX: 0px; --offsetY: -96px;"
    >
      <!---->
      <taro-view-core
        class="min-drawer-body"
      >
        <!---->
        <taro-view-core
          class="min-confirm-modal"
        >
          <!---->
          <taro-image-core
            class="icon"
            style="width: 60px; height: 60px;"
          />
          <taro-view-core
            class="title"
          >
            <!---->
            <taro-text-core
              class=""
              decode="true"
              space="nbsp"
            >
              <!---->
              Modal.confirm
            </taro-text-core>
          </taro-view-core>
          <taro-view-core
            class="subtitle"
          >
            <!---->
            <taro-text-core
              class=""
              decode="true"
              space="nbsp"
            >
              <!---->
              confirm subtitle
            </taro-text-core>
          </taro-view-core>
          <taro-view-core
            class="divider"
          >
            <!---->
          </taro-view-core>
          <taro-view-core
            class="buttons--inline"
          >
            <!---->
            <taro-button-core
              class="button cancel"
            >
              <!---->
              cancel
            </taro-button-core>
            <taro-button-core
              class="button ok"
            >
              <!---->
              ok
            </taro-button-core>
          </taro-view-core>
          <taro-view-core
            class="footer"
          >
            <!---->
            <taro-text-core
              class=""
              decode="true"
              space="nbsp"
            >
              <!---->
              confirm footer
            </taro-text-core>
          </taro-view-core>
        </taro-view-core>
      </taro-view-core>
    </taro-view-core>
  </root-portal>
</page>
`;

exports[`modal Modal.confirm render toMatchSnapshot should render Modal.confirm with cancel button 1`] = `
<page>
  <root-portal
    class="portal"
  >
    <taro-view-core
      class="min-drawer center"
      style="z-index: 1009; --duration: 300ms; --offsetX: 0px; --offsetY: -96px;"
    >
      <!---->
      <taro-view-core
        class="min-drawer-body"
      >
        <!---->
        <taro-view-core
          class="min-confirm-modal"
        >
          <!---->
          <taro-image-core
            class="icon"
            style="width: 60px; height: 60px;"
          />
          <taro-view-core
            class="title"
          >
            <!---->
            <taro-text-core
              class=""
              decode="true"
              space="nbsp"
            >
              <!---->
              Modal.confirm
            </taro-text-core>
          </taro-view-core>
          <taro-view-core
            class="subtitle"
          >
            <!---->
            <taro-text-core
              class=""
              decode="true"
              space="nbsp"
            >
              <!---->
              confirm subtitle
            </taro-text-core>
          </taro-view-core>
          <taro-view-core
            class="divider"
          >
            <!---->
          </taro-view-core>
          <taro-view-core
            class="buttons"
          >
            <!---->
            <taro-button-core
              class="button ok"
            >
              <!---->
              ok
            </taro-button-core>
            <taro-button-core
              class="button cancel"
            >
              <!---->
              cancel
            </taro-button-core>
          </taro-view-core>
          <taro-view-core
            class="footer"
          >
            <!---->
            <taro-text-core
              class=""
              decode="true"
              space="nbsp"
            >
              <!---->
              confirm footer
            </taro-text-core>
          </taro-view-core>
        </taro-view-core>
      </taro-view-core>
    </taro-view-core>
  </root-portal>
</page>
`;

exports[`modal Modal.confirm render toMatchSnapshot should render Modal.confirm with custom icon & no-divider 1`] = `
<page>
  <root-portal
    class="portal"
  >
    <taro-view-core
      class="min-drawer center"
      style="z-index: 1011; --duration: 300ms; --offsetX: 0px; --offsetY: -96px;"
    >
      <!---->
      <taro-view-core
        class="min-drawer-body"
      >
        <!---->
        <taro-view-core
          class="min-confirm-modal"
        >
          <!---->
          <taro-view-core
            class="custom-icon"
          >
            <!---->
            icon
          </taro-view-core>
          <taro-view-core
            class="title"
          >
            <!---->
            <taro-text-core
              class=""
              decode="true"
              space="nbsp"
            >
              <!---->
              Modal.confirm
            </taro-text-core>
          </taro-view-core>
          <taro-view-core
            class="no-divider"
          >
            <!---->
          </taro-view-core>
          <taro-view-core
            class="buttons"
          >
            <!---->
            <taro-button-core
              class="button ok"
            >
              <!---->
              ok
            </taro-button-core>
          </taro-view-core>
          <taro-view-core
            class="close"
          >
            <!---->
            <taro-image-core
              class=""
              style="width: 24px; height: 24px;"
            />
          </taro-view-core>
        </taro-view-core>
      </taro-view-core>
    </taro-view-core>
  </root-portal>
</page>
`;

exports[`modal Modal.with should render modal toMatchSnapshot 1`] = `
<page>
  <root-portal
    class="portal"
  >
    <taro-view-core
      class="min-drawer center"
      style="z-index: 999; --duration: 300ms; --offsetX: 0px; --offsetY: -96px;"
    >
      <!---->
      <taro-view-core
        class="min-drawer-body"
      >
        <!---->
        <taro-view-core
          class=""
          data-testid="test"
        >
          <!---->
          <taro-view-core
            class=""
          >
            <!---->
            content
          </taro-view-core>
          <taro-view-core
            class=""
          >
            <!---->
            ok
          </taro-view-core>
          <taro-view-core
            class=""
          >
            <!---->
            cancel
          </taro-view-core>
        </taro-view-core>
      </taro-view-core>
    </taro-view-core>
  </root-portal>
</page>
`;
